### PR TITLE
obs-scripting: Re-enable Python annotations; Enable Python autodoc

### PR DESCRIPTION
**Pull request by WizardCM**
### Description

Re-enables Python annotations which were automatically disabled by the upgrade to SWIG 4. swig/swig@2072ae19c
This was previously available in OBS 27.2.

Additionally enables autodoc to provide the same information to apps such as VS Code.

### Python Annotations

**Before:**

```py
def obs_property_set_long_description(p, long_description):
    return _obspython.obs_property_set_long_description(p, long_description)
```

**After:**

```py
def obs_property_set_long_description(p: "obs_property_t *", long_description: "char const *") -> "void":
    return _obspython.obs_property_set_long_description(p, long_description)

```

### Python autodoc

**Before (same with/without annotations):**

image

```py
def obs_property_set_long_description(p: "obs_property_t *", long_description: "char const *") -> "void":
    return _obspython.obs_property_set_long_description(p, long_description)

```

**After:**

image

```py
def obs_property_set_long_description(p: "obs_property_t *", long_description: "char const *") -> "void":
    r"""
    obs_property_set_long_description(p, long_description)

    Parameters
    ----------
    p: obs_property_t *
    long_description: char const *

    """
    return _obspython.obs_property_set_long_description(p, long_description)
```

### Motivation and Context

Providing more context to Python developers about what parameters they need to provide makes writing scripts faster.

### How Has This Been Tested?

Compile OBS. Check `obspython.py` and VS Code autocomplete.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by PatTheMav** [comment]
I guess this is not automatically enabled given our "stable ABI" variant.

What's the current state of type hints in SWIG? We should use valid PEP 484 hints if available, as this would enable type checking via LSP and other toolchains. Using the C-types has only visual value, proper types have code safety value.

File: deps/obs-scripting/obspython/obspython.i [hunk 1]
```diff
@@ -41,6 +41,9 @@
 
 %}
 
+%feature("python:annotations", "c");
```
Comment (PatTheMav) [inline deps/obs-scripting/obspython/obspython.i:44]: Would prefer using Python annotations or PEP 484 annotations if available.

Comment (PatTheMav) [reply deps/obs-scripting/obspython/obspython.i:44]: So only C annotations are available right now? Fair enough, we take what we can get.

Comment (WizardCM) [reply deps/obs-scripting/obspython/obspython.i:44]: Unfortunately, yeah. @CodeYan01 and I are experimenting with potential alternatives, but honestly none of them are looking fantastic.

**Pull request merged**